### DISCUSSION TITLE
Autotracking of stationary objects

### DISCUSSION THREAD

**Discussion by jakeycrx** [blakeblackshear/frigate#13664] [General]
Hi,

My PTZ tends to return the home position when the tracked object doesn't move a lot.  I know I could increase the autotracking timeout, but I prefer to leave it on 10 seconds in case it overshoots .

I do have the detect fps set to 10 fps, so would adjusting the stationary interval and/or threshold help at all?

[Example video (It was called over by a spotter)](https://drive.google.com/file/d/1QY2BK3d05TUNB1_zur6DsZKH4eFByuld/view?usp=sharing)

The scores weren't great, but I have my min score set low.
```
        cat:
          min_score: 0.5
          threshold: 0.75
```
![image](https://github.com/user-attachments/assets/e15d373f-e51b-4201-b4ff-27160f3acaa6)

**Comment by hawkeye217** [comment]
Autotracking will continue following an object, even if it becomes stationary - as long as Frigate's object tracker itself is tracking the object. So yeah, probably playing with both of those parameters would help.

**Comment by jakeycrx** [comment]
I'm still having issues with this. I have bumped my scores up a bit due to lots of training through Frigate+.
```
        cat:
          min_score: 0.6
          threshold: 0.75
```
There are 2 good examples that happened today:

[Video 1](https://drive.google.com/file/d/1MuVbcXwSD_gzSTL5qaQiSN1ot_KWPVBI/view?usp=sharing)

It looks like the score was still 63.7% when the camera was commanded to the home position at 9:44:43

![image](https://github.com/user-attachments/assets/98663594-3352-4074-85ac-dccc7c8fd327)

Object Lifecycle says the cat left at 9:44:32

![image](https://github.com/user-attachments/assets/5f680c82-2c1b-4463-98da-56120c2174c9)



[Video 2](https://drive.google.com/file/d/1KxmLYyHO1k14wN-RoRjMgM71zefG06QC/view?usp=sharing)

![image](https://github.com/user-attachments/assets/3aa71552-1abc-4ec2-972f-9ff6af75ac8e)

In this case it looks like it lost track of the cat, but it initially had a really strong score, so it seems a bit odd that it lost track of it.

![image](https://github.com/user-attachments/assets/c01f8fdb-230a-40f5-94b3-467573d0dabd)

![image](https://github.com/user-attachments/assets/15069722-f7e7-4204-8dfe-bcea5ed77535)

**Reply by hawkeye217** [reply]
If I had to guess, the issue here is probably more related to the camera movement and the scene itself than it is to the detection score of the object. The entire scene shifts, so Frigate has to calculate where an object will be after the camera moves. The number of points, edges, and contours in the image plays a huge role on how successful this calculation is. 

If you are using autotracking with zooming, the calculation is much more important. The roof doesn't have too many sharp edges and contours, so the homography calculation likely won't work well. 

If you aren't using zooming, the calculation is simpler and only uses coordinate translation. Still, a decent number of edges are required for this to be the most accurate.

This is just one of the downsides of "poor man's autotracking" and trying to make it workable for all of the hardware that Frigate can run on.

**Reply by jakeycrx** [reply]
I also have another camera (non-PTZ) set further back that also loses track when the cat sits still for a little while:

![image](https://github.com/user-attachments/assets/15016c7f-b323-41e6-9e01-8310df4a6a61)

![image](https://github.com/user-attachments/assets/c84ad929-5a07-4b81-8954-17aa67a17fb9)

**Reply by hawkeye217** [reply]
That one I'd probably guess is model related.

**Reply by NickM-27** [reply]
would be great to see the config to see what filters are set. This is likely a case where lowering the min_score a bit would probably help

**Reply by jakeycrx** [reply]
The min score is already down at 0.6
```
        cat:
          min_score: 0.6
          threshold: 0.75
```

**Reply by jakeycrx** [reply]
I'm still struggling with this even though I now have over 1600 cat images in Frigate+. I've submitted multiple images of around the time when it loses track of the cat, but it still keeps losing track when the cat is stationary.

Would it be possible to allow the return to preset timeout to be set via MQTT? That way I can increase the timeout via an automation when needed. I don't want to permanently increase it as my floodlights can cause issues with tracking objects at night, so I don't want the PTZ pointing away from a possible intruder for 30/40/50 seconds.

**Reply by hawkeye217** [reply]
That's certainly possible. Feel free to create a feature request and we'll prioritize it with the others based on user demand / number of "thumbs ups".

I've recently been working on some updates for 0.16 that will use different parameters for Frigate's internal tracker specifically for autotracking cams. That may improve your situation as well.

**Reply by jakeycrx** [reply]
How do I raise a feature request? It's something I've never done before!

I may be able to improve my setup. I currently use the default occupancy sensor from the Frigate integration to call my spotter over to the roof when a cat is detected by a fixed camera mounted at the back of the house. I'd probably be better of using my own MQTT sensor so I can fine tune it.

What are the parameters of the occupancy sensors within the integration? (score and off delay)

![image](https://github.com/user-attachments/assets/ab2e6886-4d83-4f8a-8b86-e2e7be55176c)

**Reply by hawkeye217** [reply]
There are no filters on occupancy sensors, and that is by design since some will use them for things like turning on lights, where a delay introduced by filters would be undesirable.

I'd definitely set up your own MQTT sensor to better adapt to your particular needs.

For feature requests, you can just file it here: https://github.com/blakeblackshear/frigate/issues/new/choose

**Reply by jakeycrx** [reply]
Increasing the detect resolution has made a big improvement. I was originally using the max resolution of the sub stream (1280 x 720) I'm now using the 4MP main stream transcoded to 3MP for detection (2304*1296)

**Reply by NickM-27** [reply]
Nice, glad to hear it!
